### PR TITLE
117 - Create reports lambda handler and dockerfile

### PR DESCRIPTION
**Pull request by Stefanrmc**
## Body:
Create lambda handler for summary reports generation and email system. Also create Dockerfile to build Docker image for upload to AWS ECR and deploying the lambda.

## Changes include (or breakdown of change):
- `Dockerfile` - Created dockerfile to build Docker image for AWS ECR.
- `main_reports.py` - Created helper functions and lambda handler to generate pdf reports for every station and send emails to subscribers.
- `transform_summary.py` - Added error handling for average delay functions in case a station has zero delayed trains, unlikely but possible!
- Removed `if __name__ == "__main__"` blocks from ETL scripts that I used to test functionality.

## Additional notes
Changed filenames to ensure they do not match those in other directories and fail workflow:
- `load.py` -> `load_reports.py`
- I will create a new ticket for making tests for the zero delayed trains edge case.

Closes: #117, #127.

### CODE REVIEW

File: report/main_reports.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+from load_reports import load_new_report, get_s3_client
+
+logger = logging.getLogger()
+logger.setLevel("DEBUG")
+
+
+def get_station_name_crs_tuples(conn: Connection) -> list[str]:
+    """Retrieves name and crs for each station in the database as list of tuples."""
+
+    with conn.cursor(cursor_factory=DictCursor) as curs:
```
Comment (daynyus02) [inline report/main_reports.py:26]: Can set the cursor_factory in the get_db_connection() function.

File: report/main_reports.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                    sns_client, station[1])
+                emails = get_subscriber_emails_from_topic(
+                    sns_client, topic_arn)
+
+                if emails:
+                    sent_status = send_report_emails(ses_client, emails, msg)
+                    logging.info("%s", sent_status)
+
+
+def lambda_handler(event, context) -> dict:
```
Comment (daynyus02) [inline report/main_reports.py:107]: You could do some error handling in the lambda handler just in case the run_full_email_pipeline() fails.

File: report/main_reports.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    return [sub["Endpoint"] for sub in subs["Subscriptions"]]
+
+
+def send_report_emails(ses_client: "Client", emails: list[str], msg: bytes) -> dict:
+    """Sends summary report emails to specified recipients."""
+
+    for email in emails:
+        try:
+            ses_client.send_raw_email(
+                Source="trainee.stefan.cole@sigmalabs.co.uk",
```
[the remark this answers is not in the document]
Comment (Stefanrmc) [reply report/main_reports.py:56]: It would be a nice idea to create a company email to send these alerts in the future.

File: report/main_reports.py (hunk 1 above)
Comment (Stefanrmc) [reply report/main_reports.py:26]: I found it easier to use `DictCursor` as the cursor factory to pull out the wanted data for this function rather than the default `RealDictCursor` in the get_db_connection() function.
